### ISSUE TITLE
a11y-issue: menu is a listbox

### ISSUE THREAD

**Issue by MarcusMorba** [uber/baseweb#2029]
https://baseweb.design/components/menu/

## Current Behavior
The menu component behaves like a listbox and uses listbox markup/aria (https://w3c.github.io/aria-practices/#Listbox). Also they recommended (optional) keyboard access patterns are not supported (Home/End).

## Expected Behavior
Have the tags, roles and behaviour of a menu: https://w3c.github.io/aria-practices/#menu . Support of the recommended keyboard access patterns. The menu component should also be moved to "navigation" category, instead of "pickers". The current "Menu" component should be renamed to "Listbox" within the "pickers" category.

## Your Environment
Not relevant, but Win 7 / Chrome latest.

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.7.0  |
| React   |         |
| browser |  Chrome (Win7) |

- [x] I have searched the [issues](https://github.com/uber-web/baseui/issues) of this repository and believe that this is not a duplicate.

**Comment by chasestarr** [comment]
Agreed that the component name is misleading since it does operate more like a listbox, but I think it's a mistake that we will need to accept. Consumers of baseui will not be particularly happy to update their application code because of a component name change. Do you think we could add more information to the documentation page? Please feel free to open up a PR with documentation improvements.

Can you also directly link to where page up/down should be added to a listbox? I was not able to find it in the linked page.

**Comment by MarcusMorba** [comment]
Updated and corrected the missing keyboard keys to: Home and End

A short section on the documentation page in regards of accessibility which mentions the listbox markup and behavior would help for sure.

**Comment by MarcusMorba** [comment]
Regarding the renaming: if renamed, that´s never nice for developers and should not be done in minor versions. But maybe it can be handled with a major update?

**Comment by gergelyke** [comment]
@MarcusMorba do you feel strongly about the renaming? if not, I'll close this story

**Comment by MarcusMorba** [comment]
Close it.
